storage/replica_command.go

```diff
@@ -546,12 +550,29 @@ func (r *Replica) RangeLookup(batch engine.Engine, args proto.RangeLookupRequest
 		// less than "f", and so the meta row {"f"->["c", "f")} would be
 		// ignored by MVCCReverseScan. In the second case, the range descriptor
 		// received by MVCCScan will be filtered before results are returned.
+		// Example: ranges ["c", "f") and ["f", "z"), reverse scan on ["d", "f"]
+		// leads to inspecting the descriptor {"z"->["f","z")}, which is
+		// discarded since it's not being asked for.
 		startKey, endKey := keys.MetaScanBounds(args.Key)
 		kvs, intents, err = engine.MVCCScan(batch, startKey, endKey, 1,
 			args.Timestamp, consistent, args.Txn)
 		if err != nil {
 			return reply, nil, err
 		}
+		var firstRD proto.RangeDescriptor
+		if err = gogoproto.Unmarshal(kvs[0].Value.Bytes, &firstRD); err != nil {
+			return reply, nil, err
+		}
+		if keys.RangeMetaKey(firstRD.StartKey).Less(args.Key) {
```

Comment:
yes, but it's in reverse mode so it's slightly different: "following" ranges are really the ones before (because you're scanning from high to low), and since the key you look up is the `EndKey` of the request, the first range you care for is the one that contains a key **before** `EndKey` (i.e. you want to leave out a range that you hit only because of its `StartKey`). Wanna update the corresponding comments to reflect that? I don't think they're explaining this well at the moment.